servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java

```diff
@@ -876,7 +876,9 @@ public void handlerRemoved(ChannelHandlerContext ctx) {
 
         @Override
         public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
-            connection.nettyChannelPublisher.channelOnError(unwrapThrowable(cause));
+            final Throwable throwable = unwrapThrowable(cause);
+            tryFailSubscriber(throwable);
```

Comment:
Actually, let me revert this change. It's not required for new tests in this PR and I think it's not required at all because the subscriber will be notified inside either channelActive or SSL handshake completion. If an exception is propagated before any of these 2 events, the channel will be closed anyway.
I would say we need a reproducer before we add this.